crates/ra_editor/src/symbols.rs

```diff
@@ -22,6 +22,30 @@ pub struct FileSymbol {
     pub kind: SyntaxKind,
 }
 
+impl FileSymbol {
+    pub fn docs(&self, file: &File) -> Option<String> {
+        file.syntax().descendants()
+            .filter(|node| node.kind() == self.kind && node.range() == self.node_range)
```

Comment:
Feels like we probably should move FileSymbols to ra_analysis, and use `SyntaxPtr`

Reply:
Sounds like a good idea. I can tackle that in a separate PR.